### PR TITLE
doc: Reword coding guidelines Stage 2 description

### PR DESCRIPTION
**Pull request by simhein**
Reword the Stage 2 description part about starting sometime after LTS2.
Instead a Enforcement strategy is needed beside the tooling and CI setup to get Stage 2 going

### CODE REVIEW

File: doc/contribute/coding_guidelines/index.rst [hunk 1]
```diff
@@ -16,8 +16,8 @@ Stage I
 Stage II
   Begin enforcement on a limited scope of the code base. Initially, this would be
   the safety certification scope. For rules easily applied across codebase, we
-  should not limit compliance to initial scope. This step requires tooling and
-  CI setup and will start sometime after LTS2.
+  should not limit compliance to initial scope. This step requires tooling,
+  CI setup and an enforcment strategy.
```
Comment (aasinclair) [inline doc/contribute/coding_guidelines/index.rst:20]: typo, should be 'enforcement'

Comment (simhein) [reply doc/contribute/coding_guidelines/index.rst]: thanks for the catch, fixed

**Review by aasinclair — APPROVED** [review]

**Comment by MaureenHelm** [comment]
@zephyrproject-rtos/safety please take a look

**Review by nashif — APPROVED** [review]

**Review by kestewart — APPROVED** [review]

**Comment by Thalley** [comment]
Since the timeline was removed in this PR, do we have any other timelines? 

The first paragraph still says
> Below are the agreed upon stages and the approximate timelines:

But now there are _no_ timelines or indication at what step we are at.

**Pull request merged**